lib/wab/impl/data.rb

```diff
@@ -413,33 +413,23 @@ def clone_value(value)
       end
 
       def detect_hash(h)
-        h.each_key { |k|
-          v = h[k]
-          case v
-          when Hash
-            detect_hash(v)
-          when Array
-            detect_array(v)
-          when String
-            v2 = detect_string(v)
-            h[k] = v2 unless v2 == v
-          end
-        }
+        h.each_key { |k| h[k] = detect_elememt h[k] }
```

Comment:
Better not to replace each every time. Pass in the collection (Hash or Array) along with the index/key and do the replace in the detect_element function only if the detect_string returns something other than the original.

A style thing... I like to put parenthesis around function arguments to make it clear what is being passed. Maybe it is old style. I won't reject code that does does not use parenthesis to group arguments but would if it is unclear such as ```foo 3 bar 4``` which could be ```foo(3, bar, 4)``` or ```foo(3, bar(4))```.